openwisp_controller/config/base/config.py

```diff
@@ -291,12 +291,14 @@ def _send_config_modified_signal(self):
         Emits ``config_modified`` signal.
         Called also by Template when templates of a device are modified
         """
-        config_modified.send(
-            sender=self.__class__,
-            instance=self,
-            # kept for backward compatibility
-            config=self,
-            device=self.device,
+        transaction.on_commit(
```

Comment:
Maybe this approach is not viable. A possible alternative could be to leave this as it is and add transaction.on_commit on this line:
https://github.com/openwisp/openwisp-controller/blob/8d411652a94fae120fa7998779a64fd450d32442/openwisp_controller/connection/apps.py#L44

as it's done here: https://github.com/openwisp/openwisp-firmware-upgrader/blob/4200399d3d71a0995896819f6259ebae9db9a64d/openwisp_firmware_upgrader/base/models.py#L86-L88

What happens if you try that?